packages/web-console/src/scenes/Import/ImportCSVFiles/upload-settings-dialog.tsx

```diff
@@ -0,0 +1,276 @@
+import { Settings4 } from "styled-icons/remix-line"
+import { Undo } from "styled-icons/boxicons-regular"
+import { UploadModeSettings } from "../../../utils"
+import { Drawer } from "../../../components/Drawer"
+
+const SettingsIcon = styled(Settings4)`
+  color: ${({ theme }) => theme.color.foreground};
+`
+
+const Row = styled(Box).attrs({ justifyContent: "space-between", gap: "2rem" })`
+  width: 100%;
+`
+
+const InputWrapper = styled.div`
+  flex-shrink: 0;
+  width: 30%;
+`
+
+type Props = {
+  open: boolean
+  onOpenChange: (value: boolean) => void
+  file: ProcessedFile
+  onSubmit: (settings: UploadModeSettings) => void
+}
+
+type BooleanOption = {
+  type: "switch"
+  defaultValue: boolean
+}
+
+type InputOption = {
+  type: "input"
+  defaultValue: string
+  placeholder?: string
+}
+
+type SelectOption = {
+  type: "select"
+  defaultValue: string
+  options: { label: string; value: string }[]
+}
+
+type Option = {
+  name: string
+  label: string
+  description?: React.ReactNode
+} & (BooleanOption | InputOption | SelectOption)
+
+export const UploadSettingsDialog = ({
+  file,
+  open,
+  onOpenChange,
+  onSubmit,
+}: Props) => {
+  const { delimiter, overwrite, forceHeader, skipLev, atomicity, durable } =
+    file.settings
+
+  const initialState = {
+    delimiter,
+    overwrite,
+    forceHeader,
+    skipLev,
+    atomicity,
+    durable,
+  }
+
+  const [settings, setSettings] =
+    React.useState<UploadModeSettings>(initialState)
+
+  const options: Option[] = [
+    {
+      type: "input",
+      name: "delimiter",
+      label: "Delimiter",
+      placeholder: "Automatic",
+      description: (
+        <>
+          Delimiter character. When not set, import will try to detect the
+          delimiter automatically. Since automatic delimiter detection requires
+          at least two lines (rows) to be present in the file, this parameter
+          may be used to allow single line file import.
+        </>
+      ),
+      defaultValue: settings.delimiter,
+    },
+    {
+      type: "select",
+      name: "atomicity",
+      label: "Atomicity",
+      description: (
+        <>
+          Behaviour when an error is detected in the data.
+          <br />
+          <strong>Abort</strong>: the entire file will be skipped.
+          <br />
+          <strong>Skip row</strong>: the row is skipped.
+          <br />
+          <strong>Skip column</strong>: the column is skipped.
+        </>
+      ),
+      defaultValue: settings.atomicity,
+      options: [
+        {
+          label: "Skip column",
+          value: "skipCol",
+        },
+        {
+          label: "Skip row",
+          value: "skipRow",
+        },
+        {
+          label: "Abort import",
+          value: "abort",
+        },
+      ],
+    },
+    {
+      type: "switch",
+      name: "forceHeader",
+      label: "Force header",
+      description: (
+        <>
+          When <strong>false</strong>, QuestDB will try to infer if the first
```

Comment:
Not incorrect but just to be consistent:

```suggestion
          When set to <strong>false</strong>, QuestDB will try to infer if the first
```